File: software/examples/javascript/ExampleThreshold.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change to your UID

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var hr = new Tinkerforge.BrickletHeartRate(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Get threshold callbacks with a debounce time of 10 seconds (10000ms)
        hr.setDebouncePeriod(10000);

        // Configure threshold for heart rate "greater than 100 bpm" (unit is bpm)
        hr.setHeartRateCallbackThreshold('>', 100, 0);
    }
);

// Register heart rate reached callback
hr.on(Tinkerforge.BrickletHeartRate.CALLBACK_HEART_RATE_REACHED,
    // Callback function for heart rate reached callback (parameter has unit bpm)
    function (heartRate) {
        console.log('Heart Rate: ' + heartRate + ' bpm');
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/perl/example_threshold.pl
#!/usr/bin/perl

use Tinkerforge::IPConnection;
use Tinkerforge::BrickletHeartRate;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change to your UID

# Callback subroutine for heart rate reached callback (parameter has unit bpm)
sub cb_heart_rate_reached
{
    my ($heart_rate) = @_;

    print "Heart Rate: $heart_rate bpm\n";
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $hr = Tinkerforge::BrickletHeartRate->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Get threshold callbacks with a debounce time of 10 seconds (10000ms)
$hr->set_debounce_period(10000);

# Register heart rate reached callback to subroutine cb_heart_rate_reached
$hr->register_callback($hr->CALLBACK_HEART_RATE_REACHED, 'cb_heart_rate_reached');

# Configure threshold for heart rate "greater than 100 bpm" (unit is bpm)
$hr->set_heart_rate_callback_threshold('>', 100, 0);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();
